tests/unittest/_torch/modules/moe/test_moe_comm.py

```
@@ -1586,48 +1776,113 @@ def setup_test():
+def _collect_worker_results(pending: PendingWorkerResults) -> List[dict]:
+    """Collect worker results in rank order for host-side verification."""
+    return sorted((future.result() for future in pending.futures), key=lambda r: r["rank"])
+
+
+def _drain_pending_results(pending: PendingWorkerResults):
+    """Wait for already-submitted work before propagating a verification error."""
+    for future in pending.futures:
+        try:
+            future.result()
+        except Exception:
+            pass
```

Ruff (0.15.18)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)